### ISSUE TITLE
Document how to create a storage class to consume a subvolumegroup

### ISSUE THREAD

**Issue by Madhu-1** [rook/rook#14355]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The Rook documentation does not describe how to create a storage class that consumes a subvolumegroup namespace. 

**Expected behavior:**
The Rook documentation should describe how to create a storage class that consumes a subvolumegroupl. The steps are basically:
1. Create a CephFilesystem CR
2. Create a CephFilesystemSubvolumegroup CR
3. Extract the `clusterID` from the `status` of the CephFilesystemSubvolumegroup CR:

```
kubectl -n rook-ceph get cephfilesystemsubvolumegroup -o yaml
```
```yaml
  status:
    info:
      clusterID: 80fc4f4bacc064be641633e6ed25ba7e
```
4. Set the `clusterID` in the storage class to this value instead of the name of the cluster namespace.

This could be documented in the [FilesystemSubVolumeGroup CRD topic](https://rook.io/docs/rook/latest/CRDs/Shared-Filesystem/ceph-fs-subvolumegroup-crd/)

**Comment by parth-gr** [comment]
How about automating the creation of storageClass with the clusterID, as it is a multi-step process?

We have done the same for external mode https://github.com/rook/rook/blob/master/deploy/examples/import-external-cluster.sh#L124

**Comment by Madhu-1** [comment]
we cannot create the storageclass from the rook operator (for different technical reasons) I would still like to have it as manual steps which need to be done by the user.

**Comment by travisn** [comment]
@raaizik See the [dev guide](https://rook.io/docs/rook/latest-release/Contributing/development-flow/#self-assign-issue) for the keyword to self assign, thanks for picking it up!

**Comment by brunompba** [comment]
> **Is this a bug report or feature request?**
> 
> * Bug Report
> 
> **Deviation from expected behavior:** The Rook documentation does not describe how to create a storage class that consumes a subvolumegroup namespace.
> 
> **Expected behavior:** The Rook documentation should describe how to create a storage class that consumes a subvolumegroupl. The steps are basically:
> 
> 1. Create a CephFilesystem CR
> 2. Create a CephFilesystemSubvolumegroup CR
> 3. Extract the `clusterID` from the `status` of the CephFilesystemSubvolumegroup CR:
> 
> ```
> kubectl -n rook-ceph get cephfilesystemsubvolumegroup -o yaml
> ```
> 
>   status:
>     info:
>       clusterID: 80fc4f4bacc064be641633e6ed25ba7e
> 4. Set the `clusterID` in the storage class to this value instead of the name of the cluster namespace.
> 
> This could be documented in the [FilesystemSubVolumeGroup CRD topic](https://rook.io/docs/rook/latest/CRDs/Shared-Filesystem/ceph-fs-subvolumegroup-crd/)

Hi, where exactly should one insert this "clusterID" in the storaclass YAML file?

**Comment by brunompba** [comment]
> **Is this a bug report or feature request?**
> 
> * Bug Report
> 
> **Deviation from expected behavior:** The Rook documentation does not describe how to create a storage class that consumes a subvolumegroup namespace.
> 
> **Expected behavior:** The Rook documentation should describe how to create a storage class that consumes a subvolumegroupl. The steps are basically:
> 
> 1. Create a CephFilesystem CR
> 2. Create a CephFilesystemSubvolumegroup CR
> 3. Extract the `clusterID` from the `status` of the CephFilesystemSubvolumegroup CR:
> 
> ```
> kubectl -n rook-ceph get cephfilesystemsubvolumegroup -o yaml
> ```
> 
>   status:
>     info:
>       clusterID: 80fc4f4bacc064be641633e6ed25ba7e
> 4. Set the `clusterID` in the storage class to this value instead of the name of the cluster namespace.
> 
> This could be documented in the [FilesystemSubVolumeGroup CRD topic](https://rook.io/docs/rook/latest/CRDs/Shared-Filesystem/ceph-fs-subvolumegroup-crd/)

Is it possible to reference the CephFilesystemSubVolumeGroup in the StorageClass by name instead of waiting subvolumegroup to be created and then manually setting it in clusterID ?

**Comment by travisn** [comment]
> Is it possible to reference the CephFilesystemSubVolumeGroup in the StorageClass by name instead of waiting subvolumegroup to be created and then manually setting it in clusterID ?

No, the clusterId is expected to be set to that value, but I have wondered about this design as well since the storageclass cannot be defined in advance. @Madhu-1 What would it take for the csi driver to support defining the name instead of expecting the clusterID to be set to the generated value?

**Comment by Madhu-1** [comment]
> > Is it possible to reference the CephFilesystemSubVolumeGroup in the StorageClass by name instead of waiting subvolumegroup to be created and then manually setting it in clusterID ?
> 
> No, the clusterId is expected to be set to that value, but I have wondered about this design as well since the storageclass cannot be defined in advance. [@Madhu-1](https://github.com/Madhu-1) What would it take for the csi driver to support defining the name instead of expecting the clusterID to be set to the generated value?

Its not possible to put this in the storageclass(we are putting more and more configurations to the SC and not all details can be added to the volumeHandle(this is the only thing we can for further CSI operations). that why its designed in this way in ceph-csi where will look for more details from the cluster ID.

**Comment by brunompba** [comment]
Thank you for replying. Someday, if that turns out to be possible, I would grateful .

**Comment by Madhu-1** [comment]
@travisn How about providing an option to specify the clusterID in the spec field so that the admin will get the expected result and they dont need to wait for the clusterID auto generated one from Rook?

**Comment by travisn** [comment]
> [@travisn](https://github.com/travisn) How about providing an option to specify the clusterID in the spec field so that the admin will get the expected result and they dont need to wait for the clusterID auto generated one from Rook?

Sounds like a good feature to me, as per my [earlier comment](https://github.com/rook/rook/issues/14355#issuecomment-2784287674).
